src/ServiceDefinitionInterface.php

```diff
@@ -0,0 +1,16 @@
+<?php
+
+namespace Interop\Container;
+
+/**
+ * Represents a service definition.
+ */
+interface ServiceDefinitionInterface
+{
+    /**
+     * Retrieves the keys of known dependencies of this service.
+     *
+     * @return string[] A list of service keys.
+     */
+    public function getDependencies(): array;
+}
```

Comment:
The ability to have dependencies is indeed a trait of all service definitions. However, a service definition is primarily defined as something that can be resolved to a service, and the dependencies part seems secondary to that. So while splitting the `Service` facet into `Factory` and `Extension` is an elegant way to separate concerns, I'm not sure whether it represents the nature of the problem.

Reply:
It wasn't done that way to separate concerns - it was just a means of reuse.

Do you think this would make more sense as a `trait`?

Reply:
Let's leave this detail for later.

I think the importing part for now is we incorporate support for dependency enumeration - I don't expect we'll get this right the first time, and I'm sure we will make more changes before we publish a release or start attempting to code against these interfaces.